packages/babel-plugin-transform-async-generator-functions/test/fixtures/for-await/async-generator-exec/exec.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+async function* genAnswers() {
+  var stream = [ Promise.resolve(4), Promise.resolve(9), Promise.resolve(12) ];
+  var total = 0;
+  for await (let val of stream) {
+    total += await val;
+    yield total;
+  }
+}
+
+function forEach(ai, fn) {
+  return ai.next().then(function (r) {
+    if (!r.done) {
+      fn(r);
+      return forEach(ai, fn);
+    }
+  });
+}
+
+var output = 0;
+return forEach(genAnswers(), function(val) { output += val.value })
+.then(function () {
+  assert.equal(output, 42);
```

Comment:
`assert.equal(output, 42);` in a promise. Needs to use the `allowReturnOutsideFunction` parser option